vmicore/src/lib/plugins/PluginSystem.cpp

```diff
@@ -161,17 +161,19 @@ namespace VmiCore
                                               Plugin::PluginInterface::API_VERSION));
         }
 
-        auto pluginInitFunction = std::bit_cast<decltype(Plugin::init)*>(
-            dlsym(libraryHandle,
-                  "_ZN7VmiCore6Plugin4initEPNS0_15PluginInterfaceENSt3__110shared_ptrINS0_13IPluginConfigEEENS3_"
-                  "6vectorINS3_12basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEENSB_ISD_EEEE"));
+        auto pluginInitFunction =
+            std::bit_cast<decltype(Plugin::init_plugin)*>(dlsym(libraryHandle, PLUGIN_INIT_FUNCTION.data()));
         dlErrorMessage = dlerror();
         if (dlErrorMessage != nullptr)
         {
             throw PluginException(pluginName, fmt::format("Unable to retrieve init function: {}", dlErrorMessage));
         }
 
         auto plugin = pluginInitFunction(dynamic_cast<Plugin::PluginInterface*>(this), std::move(config), args);
+        if (plugin == nullptr)
+        {
+            throw PluginException(pluginName, "init function returned null");
+        }
```

Comment:
I think you forgot to remove this? The init function can never return a nullptr, just an empty unique_ptr

Reply:
Yikes, thanks